File: src/main/java/me/alegian/thavma/impl/init/registries/deferred/T7Attributes.java
package me.alegian.thavma.impl.init.registries.deferred;

import me.alegian.thavma.impl.Thavma;
import net.minecraft.Util;
import net.minecraft.core.registries.Registries;
import net.minecraft.world.entity.ai.attributes.Attribute;
import net.neoforged.neoforge.common.BooleanAttribute;
import net.neoforged.neoforge.registries.DeferredHolder;
import net.neoforged.neoforge.registries.DeferredRegister;

public class T7Attributes {
  public static final DeferredRegister<Attribute> REGISTRAR = DeferredRegister.create(Registries.ATTRIBUTE, Thavma.MODID);

  public static final DeferredHolder<Attribute, Attribute> REVEALING = REGISTRAR.register("revealing", (key) ->
      new BooleanAttribute(
          Util.makeDescriptionId(Registries.ATTRIBUTE.location().getPath(), key),
          false
      ).setSyncable(true)
  );
}

File: src/main/java/me/alegian/thavma/impl/init/registries/deferred/T7Attributes.kt
package me.alegian.thavma.impl.init.registries.deferred

import me.alegian.thavma.impl.Thavma
import net.minecraft.Util
import net.minecraft.core.registries.Registries
import net.minecraft.world.entity.ai.attributes.Attribute
import net.neoforged.neoforge.common.BooleanAttribute
import net.neoforged.neoforge.registries.DeferredHolder
import net.neoforged.neoforge.registries.DeferredRegister

object T7Attributes {
    val REGISTRAR: DeferredRegister<Attribute> = DeferredRegister.create(Registries.ATTRIBUTE, Thavma.MODID)
    
    val REVEALING: DeferredHolder<Attribute, Attribute> = REGISTRAR.register(
        "revealing"
    ) { rl ->
        BooleanAttribute(
            Util.makeDescriptionId(Registries.ATTRIBUTE.location().path, rl),
            false
        ).setSyncable(true)
    }
}
